tests/ModelContextProtocol.Tests/Client/McpClientExtensionsTests.cs

```diff
@@ -171,4 +171,32 @@ public async Task GetPromptsAsync_HonorsJsonSerializerOptions()
 
         await Assert.ThrowsAsync<NotSupportedException>(() => client.GetPromptAsync("Prompt", new Dictionary<string, object?> { ["i"] = 42 }, emptyOptions, cancellationToken: TestContext.Current.CancellationToken));
     }
+
+    [Fact]
+    public async Task WithName_ChangesToolName()
+    {
+        JsonSerializerOptions options = new(JsonSerializerOptions.Default);
+        IMcpClient client = await CreateMcpClientForServer();
+
+        var tool = (await client.ListToolsAsync(options, TestContext.Current.CancellationToken)).FirstOrDefault();
+        var originalName = tool?.Name;
+        var renamedTool = tool?.WithName("RenamedTool");
```

Comment:
Do we expect the client fixture to not return tools?
```suggestion
        var tool = (await client.ListToolsAsync(options, TestContext.Current.CancellationToken)).First();
        var originalName = tool.Name;
        var renamedTool = tool.WithName("RenamedTool");
```